sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -304,46 +308,68 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
-    }
+    import df.sparkSession.sessionState.analyzer.CatalogAndIdentifier
 
-    if (source == SOURCE_NAME_TABLE) {
-      assertNotPartitioned(SOURCE_NAME_TABLE)
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+    val originalMultipartIdentifier = df.sparkSession.sessionState.sqlParser
+      .parseMultipartIdentifier(tableName)
+    val CatalogAndIdentifier(catalog, identifier) = originalMultipartIdentifier
 
-      import df.sparkSession.sessionState.analyzer.CatalogAndIdentifier
+    // Currently we don't create a logical streaming writer node in logical plan, so cannot rely
+    // on analyzer to resolve it. Directly lookup only for temp view to provide clearer message.
+    // TODO (SPARK-27484): we should add the writing node before the plan is analyzed.
+    if (df.sparkSession.sessionState.catalog.isTempView(originalMultipartIdentifier)) {
+      throw new AnalysisException(s"Temporary view $tableName doesn't support streaming write")
+    }
 
+    if (!catalog.asTableCatalog.tableExists(identifier)) {
       import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
-      val originalMultipartIdentifier = df.sparkSession.sessionState.sqlParser
-        .parseMultipartIdentifier(tableName)
-      val CatalogAndIdentifier(catalog, identifier) = originalMultipartIdentifier
-
-      // Currently we don't create a logical streaming writer node in logical plan, so cannot rely
-      // on analyzer to resolve it. Directly lookup only for temp view to provide clearer message.
-      // TODO (SPARK-27484): we should add the writing node before the plan is analyzed.
-      if (df.sparkSession.sessionState.catalog.isTempView(originalMultipartIdentifier)) {
-        throw new AnalysisException(s"Temporary view $tableName doesn't support streaming write")
-      }
+      val cmd = CreateTableStatement(
+        originalMultipartIdentifier,
+        df.schema.asNullable,
+        partitioningColumns.getOrElse(Nil).asTransforms.toSeq,
+        None,
+        Map.empty[String, String],
+        Some(source),
+        Map.empty[String, String],
+        extraOptions.get("path"),
+        None,
+        None,
+        external = false,
+        ifNotExists = false)
+      Dataset.ofRows(df.sparkSession, cmd)
+    }
 
-      val tableInstance = catalog.asTableCatalog.loadTable(identifier)
+    val tableInstance = catalog.asTableCatalog.loadTable(identifier)
 
-      import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Implicits._
-      val sink = tableInstance match {
-        case t: SupportsWrite if t.supports(STREAMING_WRITE) => t
-        case t => throw new AnalysisException(s"Table $tableName doesn't support streaming " +
-          s"write - $t")
-      }
+    def writeToV1Table(table: CatalogTable): StreamingQuery = {
+      require(table.tableType != CatalogTableType.VIEW, "Streaming into views is not supported.")
+      format(table.provider.get)
```

Comment:
Writing to V1 table shouldn't rely on `normalizedParCols`, but then you'll be stuck how to provide `Array[Transform]` (provided by CatalogTable) to `Seq[String]`.

Reply:
I see `ResolveSessionCatalog.buildCatalogTable()` leverages `partitioning.asPartitionColumns` - that could be used instead.

Reply:
I think we shouldn't set `partitionBy` here since that will ignore the original setting of partitioningColumns.

Reply:
So are we completely ignoring the table's partition information? I don't think this is same as DSv2 path.

Reply:
We didn't ignore. IMO, the partition information should be handled at the DataSource level, no matter V1/V2. Different DS should have their own ability and strategy to handle the partition overwrite/append issue(part discussion of schema evolution). So for the API level, we need to pass both information down to the runtime.

Reply:
Please find the usage of `normalizedParCols` in DataStreamWriter. This config is only effective in DSv1.

Reply:
Got the point. So your point focus on how DSv2 addressing the partition column conflict. My point is want to explain we shouldn't overwrite the user's input, we need to pass both(user-input and table catalog partition info) into the data source. I'll add this and further discussion to SPARK-33638 since it's part of full support for the V2 table.

Reply:
Sorry I missed this. I'm not sure we provide the both into data source, but data source is probably able to know about the partitioning (as table partitioning is given by data source), so consider this as minor and make a follow-up if necessary. In anyway you'll want to check this to achieve my review comments on documentation.

Reply:
`I'm not sure we provide the both into data source`
My bad, didn't make it clear. We don't provide both, only pass the user-provided one(in V1). The partitioning in the catalog is able to know in a data source as you said. I need to change `we need to pass both` to `data source need to know both(we need to pass the user-provided partitioning)`. Let me rephrase the last comment to make it clear.

Yes, of cause, a follow-up is needed.